spec/models/user_spec.rb

```diff
@@ -416,27 +416,67 @@
   end
 
   describe "self.search" do
-    it "find users by email" do
-      user1 = create(:user, email: "larry@consul.dev")
-      create(:user, email: "bird@consul.dev")
-      search = User.search("larry@consul.dev")
+    describe "find users" do
+      let!(:larry) { create(:user, email: "larry@consul.dev", username: "Larry Bird") }
 
-      expect(search).to eq [user1]
-    end
+      before { create(:user, email: "robert@consul.dev", username: "Robert Parish") }
+
+      it "by email" do
+        expect(User.search("larry@consul.dev")).to eq [larry]
+      end
 
-    it "find users by name" do
-      user1 = create(:user, username: "Larry Bird")
-      create(:user, username: "Robert Parish")
-      search = User.search("larry")
+      it "by email with whitespaces" do
+        expect(User.search(" larry@consul.dev ")).to eq [larry]
+      end
+
+      it "by name" do
+        expect(User.search("larry")).to eq [larry]
+      end
 
-      expect(search).to eq [user1]
+      it "by name with whitespaces" do
+        expect(User.search(" larry ")).to eq [larry]
+      end
     end
 
     it "returns no results if no search term provided" do
       expect(User.search("    ")).to be_empty
     end
   end
 
+  describe "scope" do
+    describe ".by_username_email_or_document_number" do
+      let!(:larry) do
+        create(:user, email: "larry@consul.dev", username: "Larry Bird", document_number: "12345678Z")
+      end
+
+      before { create(:user, email: "robert@consul.dev", username: "Robert Parish") }
+
+      it "find users by email" do
```

Comment:
Typo: use the third person after an `it` :wink:. Otherwise reading this line feels strange.

```suggestion
      it "finds users by email" do
```

Same in the other tests.